### ISSUE TITLE
Slim models: How to know model-specific std and mean values

### ISSUE THREAD

**Issue by Johnson145** [tensorflow/models#3346]
### Describe the problem
I'm wondering how to know which mean and std values should be used for running inference with a specific net.

For instance, [this](https://github.com/tensorflow/models/tree/master/research/slim#run-label-image-in-c) suggests that the slim inception-v3 model requires `input_mean=0` and `input_std=255` to be used. However, the "old non-slim" inception-v3 model used `input_mean=128` and `input_std=128` ([source](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/examples/image_retraining/retrain.py)). 

I see that the first one normalizes all inputs from [0, 255] to [0, 1], while the latter maps all values to [-1, 1]. Of course, both choices can make sense as long as the same choice was already used during the training process. However, I don't understand where to find this choice in the training script. Can someone point me to the correct slim code part that implies the given values?

PS:
This may be a question for Stack Overflow as well. However, as these values are required to successfully deploy a net, I think this info should be added to the docs, too.

### System information
(not relevant here)
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**:
N/A
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**:
N/A
- **TensorFlow installed from (source or binary)**:
N/A
- **TensorFlow version (use command below)**:
N/A
- **Python version**: 
N/A
- **Bazel version (if compiling from source)**:
N/A
- **GCC/Compiler version (if compiling from source)**:
N/A
- **CUDA/cuDNN version**:
N/A
- **GPU model and memory**:
N/A
- **Exact command to reproduce**:
N/A

**Comment by Johnson145** [comment]
I guess I found an answer at least for the inception-v3 net:

https://github.com/tensorflow/models/blob/fd7b6887fb294e90356d5664724083d1f61671ef/research/slim/preprocessing/inception_preprocessing.py#L254-L257

Wouldn't it make sense to include the `tf.image.convert_image_dtype` method into the finally exported net as well? Such that we do not need to care about manual normalization during inference?

**Comment by Johnson145** [comment]
I'm still confused about this. Let's limit this discussion to the inception-v3 example for now.

As described in my previous comment, the inception preprocessing methods do accept values in [0, 1] (or more generally in [0, MAX]). If this was our entry point, we will need to apply normalization with 0/255.

However, the preprocessing method isn't embedded into the graph when exporting it. If we take a closer look at the end of the preprocessing steps, we see that the following lines
https://github.com/tensorflow/models/blob/d1c48afcd0ca503bf8a320cfe862eed04217c68d/research/slim/preprocessing/inception_preprocessing.py#L280-L282
do shift the actual net input from [0, 1] to [-1, 1]. So the actual net input (of an exported graph) must be in [-1, 1], too, right? So we actually still need to normalize our input with 128/128 instead of 0/255, don't we?

**Afaik the latter is in contrast to all official TensorFlow examples using the deployed inception net. Are all those docs wrong, or am I missing something here?**

**Comment by Johnson145** [comment]
To continue my monologue:

I've just confirmed by assumption experimentally. I've evaluate both parameter configuration for 1M images and indeed, the 128/128 combination results in much better accuracy values than the suggested 0/255 one does.

**Comment by dkashkin** [comment]
Can we please get some more clarity on this from the Tensorflow team? The choice of MEAN and STD values impacts the prediction results in a very dramatic way, so it would be super helpful to have some official guidance on how we should treat these parameters...

**Comment by tensorflowbutler** [comment]
Hi There,
 We are checking to see if you still need help on this, as this seems to be considerably old issue. Please update this issue with the latest information, code snippet to reproduce your issue and error you are seeing.
 If we don't hear from you in the next 7 days, this issue will be closed automatically. If you don't need help on this issue any more, please consider closing this.
